packages/medusa-plugin-sendgrid/src/services/sendgrid.js

```diff
@@ -304,6 +311,9 @@ class SendGridService extends NotificationService {
    * @return {Promise} result of the send operation
    */
   async sendEmail(options) {
+    if (!this.options_.enable_endpoint) { 
```

Comment:
todo: I feels weird to have to enable_endpoint if you want to use the service method somewhere. Should we have that in the endpoint instead?

Reply:
It's a little strange, but this method is not the main sending method.  It is a copy of the method that is used solely with the API endpoint. Disabling the endpoint by disabling this method does not affect the normal functioning of notifications at all.

We could edit the endpoint to also check for whether the option is enabled, and if not return a message that it has been disabled.  That would alert users who may be currently using the endpoint.   It would also have them notice that they should only be using it in a secure testing environment, or, if they really think they need it in prod, they should be securing it in some way, most likely via their reverse proxy configuration.

Reply:
Just to document rationale: I would rather not add authenticate() middleware, because I think it is really important for people to be able to test templates **very easily** without having to trigger the actual medusa events and without having to do a lot of set up.  

Disabling by default and providing a warning about security I think is a better approach.

Reply:
I know the solution I proposed is extremely basic, but I think it is sufficient in this case, and patching this sooner rather than later is probably best.

Reply:
I understand your point, but as far as I can see, if the server is deployed with the option enabled, then any admin can still send emails plus any users that find a right template id, correct me if I am wrong.

Reply:
also, if the option is disabled, you can't send any email from the server in the places that are using this method. So in the end this option completely disabled the plugin from being able to send an email from within a project.

Let's say that a user installs this plugin, and creates a new subscriber that on a certain event sends an email. If the option is disabled, the email will not be sent and the admin won't have any error to see that the emails are not sent, and if the user enables the option then the subscriber can send the email but the endpoint is still accessible by anybody with a valid template id

Reply:
The function returns, meaning it stops.  So it is the easiest way to make the endpoint do nothing.  I have used this method with the ses plugin since it was released, and it is blunt but effective.

As mentioned above, this function within the service is used absolutely no where else except in the API endpoint.

Reply:
I agree that it is used nowhere than in the end point. But the fact is that the public API of that service makes that method available to the consumer if he wants to use it elsewhere. What I mean is that it is not because the method is not used elsewhere than in the endpoint, than the user does not use it anywhere since the user can use it anywhere he wants.

Reply:
It's true, someone may have used it.  Hopefully, in most cases, they used the more robust function.

I think issues could be mitigated by prominently noting in the changelog and bumping the minor version to prevent automatic upgrade for most people.